src/main/java/jenkins/plugins/office365connector/FactsBuilder.java

```diff
@@ -118,6 +126,12 @@ private Collection sortUsers(Set<User> authors) {
                 .collect(Collectors.toList());
     }
 
+    public void addCustom(String message, Run run) throws IOException, InterruptedException {
+        EnvVars envVars;
+        envVars = run.getEnvironment(new LogTaskListener(LOGGER, Level.INFO));
+        addFact(ADDITIONAL_INFO, envVars.expand(message));
```

Comment:
I think you can reuse method that processes macro which I believe produces the same results